jason-lab/server.js

```diff
@@ -0,0 +1,96 @@
+'use strict';
+
+const net = require('net');
+const PORT = process.env.port || 3000;
+const server = net.createServer();
+
+const EE = require('events');
+const ee = new EE();
+
+const ansi = require('ansi')
+  , cursor = ansi(process.stdout);
+
+const pool = [];
+const uuid = require('uuid/v4');
+
+const helpMenu = `Type @nick folowed by a nickname of your choice to change your nickname
+\n\t- nicknames can not have spaces -
+\n\t- nicknames can contain letters, numbers, or a combination of both -
+\t\n Type @close to exit.
+\t\n Type @dm followed my a User ID or nickname to direct message a user\n`;
+
+function Client(socket){
+  this.nickname = 'guest';
+  this.socket = socket;
+  this.id = uuid();
+}
+
+server.listen(PORT, function(){
+  console.log(`server is listening on ${PORT}`);
+});
+
+ee.on('@all', function (sender, message) {
+  pool.forEach(receiver => {
+    if (receiver.id === sender.id) return;
+    receiver.socket.write(`${sender.nickname}: ${message}`);
+  });
+});
+
+ee.on('@dm', function (sender, string){
+  let nickname = string.split(' ').shift().cursor.orange();
```

Comment:
Similar to `toString()`, it's generally best to avoid repeating a `split()`.

Also, naming a variable after a type (`string`) is usually an indication that you're missing an opportunity to communicate what the variable is.